neo/io/kwikio.py

```diff
@@ -0,0 +1,206 @@
+        except:
+            self._attrs['app_data'] = False
+
+    def read_block(self,
+                     lazy=False,
+                     cascade=True,
+                     channel_index=None
+                    ):
+        """
+        Arguments:
+            Channel_index: can be int, iterable or None to select one, many or all channel(s)
+
+        """
+
+        blk = Block()
+        if cascade:
+            seg = Segment( file_origin=self._filename )
+            blk.segments += [ seg ]
+
+
+
+            if channel_index:
+                if type(channel_index) is int: channel_index = [ channel_index ]
+                if type(channel_index) is list: channel_index = np.array( channel_index )
+            else:
+                channel_index = np.arange(0,self._attrs['shape'][1])
+
+            rcg = RecordingChannelGroup(name='all channels',
+                                 channel_indexes=channel_index)
+            blk.recordingchannelgroups.append(rcg)
+
+            for idx in channel_index:
+                # read nested analosignal
+                ana = self.read_analogsignal(channel_index=idx,
+                                        lazy=lazy,
+                                        cascade=cascade,
+                                         )
+                chan = RecordingChannel(index=int(idx))
+                seg.analogsignals += [ ana ]
+                chan.analogsignals += [ ana ]
+                rcg.recordingchannels.append(chan)
+            seg.duration = (self._attrs['shape'][0]
+                          / self._attrs['kwik']['sample_rate']) * pq.s
+
+            # neo.tools.populate_RecordingChannel(blk)
+        blk.create_many_to_one_relationship()
+        return blk
+
+    def read_analogsignal(self,
+                      channel_index=None,
+                      lazy=False,
+                      cascade=True,
+                      ):
+        """
+        Read raw traces
+        Arguments:
+            channel_index: must be integer
+        """
+        try:
+            channel_index = int(channel_index)
+        except TypeError:
+            print('channel_index must be int, not %s' %type(channel_index))
+
+        if self._attrs['app_data']:
+            bit_volts = self._attrs['app_data']['channel_bit_volts']
+            sig_unit = 'uV'
+        else:
+            bit_volts = np.ones((self._attrs['shape'][1])) # TODO: find conversion in phy generated files
+            sig_unit =  'bit'
+        if lazy:
+            anasig = AnalogSignal([],
+                                  units=sig_unit,
+                                  sampling_rate=self._attrs['kwik']['sample_rate']*pq.Hz,
+                                  t_start=self._attrs['kwik']['start_time']*pq.s,
+                                  channel_index=channel_index,
+                                  )
+            # we add the attribute lazy_shape with the size if loaded
+            anasig.lazy_shape = self._attrs['shape']
```

Comment:
I  guess it is anasig.lazy_shape = self._attrs['shape'][0]
self._attrs['shape'] is ndim==2 no ?